app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt

```diff
@@ -592,22 +594,22 @@ fun GalleryDetailScreen(args: GalleryDetailScreenArgs, navigator: DestinationsNa
             val downloadTorrentFailed = stringResource(R.string.download_torrent_failure)
             val downloadTorrentStarted = stringResource(R.string.download_torrent_started)
             val noTorrents = stringResource(R.string.no_torrents)
-            var mTorrentList by remember { mutableStateOf<TorrentResult?>(null) }
-            suspend fun showTorrentDialog() {
-                if (mTorrentList == null) {
-                    mTorrentList = bgWork {
-                        withIOContext {
-                            EhEngine.getTorrentList(galleryDetail.torrentUrl!!, gid, token)
-                        }
+            val torrentList = remember {
+                async(Dispatchers.IO, CoroutineStart.LAZY) {
+                    runSuspendCatching {
+                        EhEngine.getTorrentList(galleryDetail.torrentUrl!!, gid, token)
                     }
                 }
-                if (mTorrentList!!.isEmpty()) {
+            }
+            suspend fun showTorrentDialog() {
+                val list = bgWork { torrentList.await() }.getOrThrow()
+                if (list.isEmpty()) {
                     showSnackbar(noTorrents)
```

Comment:
nit: May change this to something like "No torrents for the current version (of the gallery)".